modules/Kconfig.s32

```diff
@@ -1,9 +1,13 @@
 # NXP S32 MCUs family
 
-# Copyright 2022 NXP
+# Copyright 2022-2023 NXP
 # SPDX-License-Identifier: Apache-2.0
 
 config HAS_S32_HAL
 	bool
 	select HAS_CMSIS_CORE
 	depends on SOC_FAMILY_S32
+
+config NOCACHE_LOAD_MEMORY
```

Comment:
perhaps we could consider making this available to other SoCs? similarly to `CONFIG_NOCACHE_MEMORY`

Reply:
Can you add a description for this Kconfig

Reply:
When i revisited the bare-metal driver and had a discussion with the engineer from HAL team, he said that actually the variable is supposed and can be put in to a read only region. We will revisit the whole supporting features and if needed, will put a separate PR for this support